File: java/app-encryption/src/test/java/com/godaddy/asherah/appencryption/DefaultPartitionTest.java
package com.godaddy.asherah.appencryption;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class DefaultPartitionTest {

    private final static String testPartitionId = "test_partition_id";
    private final static String testServiceId = "test_service_id";
    private final static String testProductId = "test_product_id";

    private Partition partition;

    @BeforeEach
    void setUp() {
        partition = new DefaultPartition(testPartitionId, testServiceId, testProductId);
    }

    @Test
    void testGetSystemKeyId() {
        String systemKeyIdString = "_SK_" + testServiceId + "_" + testProductId;
        assertEquals(systemKeyIdString, partition.getSystemKeyId());
    }

    @Test
    void testGetIntermediateKeyId() {
        String intermediateKeyIdString = "_IK_" + testPartitionId + "_" + testServiceId + "_" + testProductId;
        assertEquals(intermediateKeyIdString, partition.getIntermediateKeyId());
    }

    @Test
    void testToString() {
        String toStringString = partition.getClass().getSimpleName() + "[partitionId=" + testPartitionId +
                ", serviceId=" + testServiceId + ", productId=" + testProductId + "]";
        assertEquals(toStringString, partition.toString());
    }
}

File: csharp/AppEncryption/AppEncryption.Tests/AppEncryption/DefaultPartitionTest.cs
using Xunit;

namespace GoDaddy.Asherah.AppEncryption.Tests.AppEncryption
{
  [Collection("Logger Fixture collection")]
  public class DefaultPartitionTest
  {
    private const string TestPartitionId = "test_partition_id";
    private const string TestServiceId = "test_service_id";
    private const string TestProductId = "test_product_id";

    private readonly DefaultPartition partition;

    public DefaultPartitionTest()
    {
      partition = new DefaultPartition(TestPartitionId, TestServiceId, TestProductId);
    }

    [Fact]
    private void TestPartitionCreation()
    {
      Assert.NotNull(partition);
    }

    [Fact]
    private void TestGetPartitionId()
    {
      string actualTestPartitionId = partition.PartitionId;
      Assert.Equal(TestPartitionId, actualTestPartitionId);
    }

    [Fact]
    private void TestGetServiceId()
    {
      string actualServiceId = partition.ServiceId;
      Assert.Equal(TestServiceId, actualServiceId);
    }

    [Fact]
    private void TestGetProductId()
    {
      string actualProductId = partition.ProductId;
      Assert.Equal(TestProductId, actualProductId);
    }

    [Fact]
    private void TestGetSystemKeyId()
    {
      const string systemKeyIdString = "_SK_" + TestServiceId + "_" + TestProductId;
      Assert.Equal(systemKeyIdString, partition.SystemKeyId);
    }

    [Fact]
    private void TestGetIntermediateKeyId()
    {
      const string intermediateKeyIdString = "_IK_" + TestPartitionId + "_" + TestServiceId + "_" + TestProductId;
      Assert.Equal(intermediateKeyIdString, partition.IntermediateKeyId);
    }

    [Fact]
    private void TestToString()
    {
      string expectedToStringString = partition.GetType().Name + "[partitionId=" + TestPartitionId +
              ", serviceId=" + TestServiceId + ", productId=" + TestProductId + "]";
      Assert.Equal(expectedToStringString, partition.ToString());
    }

    [Fact]
    private void TestIsValidIntermediaKeyId()
    {
      const string intermediateKeyIdString = "_IK_" + TestPartitionId + "_" + TestServiceId + "_" + TestProductId;
      Assert.True(partition.IsValidIntermediateKeyId(intermediateKeyIdString));

      const string invalidId = "_IK_some_other_partition" + "_" + TestServiceId + "_" + TestProductId;
      Assert.False(partition.IsValidIntermediateKeyId(invalidId));
    }
  }
}
